### PR TITLE
Handle case-insensitive sensitive headers

### LINKED ISSUE #673: SILKY_HIDE_COOKIES does not hide cookie header in different case
The SILKY_HIDE_COOKIES setting does not hide the cookie header if it is written in a different case.

**Current behavior:**
When using the `SILKY_HIDE_COOKIES` setting, the cookie header is not hidden if it is written in a different case compared to the specified name in the code in upper case 
**Expected behavior:**
The `SILKY_HIDE_COOKIES` setting should hide the cookie header, regardless of the case in which it is written. Currently, it only works if the case matches exactly.

**Steps to reproduce:**
1. Set the `SILKY_HIDE_COOKIES` setting to True.
2. Send a request with a 'cookie' header in lower case.
3. Inspect the request headers on the server side, and observe that the 'COOKIE' header is present.

**Refs**
- `COOKIE` in upper case is hardcoded [here](https://github.com/jazzband/django-silk/blob/master/silk/model_factory.py#L79)

### PR DESCRIPTION
**Pull request by shtimn**
fixes #673

### CODE REVIEW

File: silk/model_factory.py [hunk 1]
```diff
@@ -66,19 +66,20 @@ def encoded_headers(self):
         """
         From Django docs (https://docs.djangoproject.com/en/2.0/ref/request-response/#httprequest-objects):
         """
-        headers = {}
-        sensitive_headers = {'authorization'}
+        sensitive_headers = SilkyConfig().SILKY_SENSITIVE_KEYS
```
Comment (kekekekule) [inline silk/model_factory.py:69]: maybe it's worth using 
```
set(SilkyConfig().SILKY_SENSITIVE_KEYS)
```
just for right type guarantee

File: silk/model_factory.py [hunk 2]
```diff
@@ -66,19 +66,20 @@ def encoded_headers(self):
...
+
+        sensitive_headers.add('authorization')
+        if SilkyConfig().SILKY_HIDE_COOKIES:
+            sensitive_headers.add('cookie')
+
+        sensitive_headers = map(str.lower, sensitive_headers)
 
+        headers = {}
         for k, v in self.request.headers.items():
             if k in sensitive_headers:
```
Comment (kekekekule) [inline silk/model_factory.py:79]: let's do k.lower() maybe to avoid issues, if smth changes in Django?

**Review by kekekekule — CHANGES_REQUESTED** [review]

File: silk/model_factory.py (hunk 2 above)
Comment (shtimn) [reply silk/model_factory.py:80]: Oh, really
I missed that

File: silk/model_factory.py (hunk 1 above)
Comment (shtimn) [reply silk/model_factory.py]: you're right
done

**Review by kekekekule — APPROVED** [review]

**Comment by shtimn** [comment]
Hi @albertyw,

I have made some improvements to the handling of sensitive headers in the `RequestModelFactory.encoded_headers` method. 
Could you please take a moment to review the changes ? 

Thanks!

**Review by albertyw — APPROVED** [review]

**Comment by shtimn** [comment]
@albertyw,  I appreciate the merge 😊

Could you please let me know when the new version is scheduled?

**Comment by albertyw** [comment]
New version will be released with #678

**Pull request merged**